File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int element = 0, count = 0;
        for(int i = 0; i < nums.length; i++){
            if(count == 0){
                count++;
                element = nums[i];
            } else if(element == nums[i]){
                count++;
            } else{
                count--;
            }
        }

        int newCount = 0;
        for(int i = 0; i < nums.length; i++){
            if(element == nums[i]){
                newCount++;
            }
        }
        if(newCount > nums.length/2){
            return element;
        }
        return -1;
    }
}

File: 0169-majority-element/0169-majority-element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function (nums) {
    let count = 0;
    let elem = nums[0];
    for (let i = 0; i < nums.length; i++) {
        if (elem === nums[i]){
            count++;
        } else if(count === 0){
            elem = nums[i];
        } else {
            count--;
        }
    }

    return elem;
};